tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/UpToDate/BuildUpToDateCheckTests.cs

```diff
@@ -1095,6 +1095,36 @@ public async Task IsUpToDateAsync_False_CopyToOutputDirectoryDestinationDoesNotE
                 "CopyToOutputDirectory");
         }
 
+        [Fact]
+        public async Task IsUpToDateAsync_True_AdditionalDependentFilesIgnoredWhenCountIsZero()
+        {
+            var projectSnapshot = new Dictionary<string, IProjectRuleSnapshotModel>
+            {
+                [UpToDateCheckBuilt.SchemaName] = SimpleItems("BuildDefault"),
+                [UpToDateCheckInput.SchemaName] = ItemWithMetadata("Input1", "Set", "Set1"),
+            };
+
+            var dependentTime = DateTime.UtcNow.AddMinutes(-5);
```

Comment:
I think the test would be more conclusive if `dependentTime` was the newest time of all, as we're trying to verify it's definitely ignored.

Reply:
Update() only checks when files are added or removed. It doesn't check for file changes. 
I will add the change suggested in next commit.

Reply:
The bit that I think is missing from test coverage of this feature is a `BroadcastChange` that tests the internal machinery for updating the state in response to external changes. [`InitialItemDataDoesNotUpdateLastItemsChangedAtUtc`](https://github.com/dotnet/project-system/blob/a33325762bdf3eb810ad71def2a15d9839a9631f/tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/UpToDate/BuildUpToDateCheckTests.cs#L479) is a good example of that. It may be that you need another test, rather than to change this one.

Reply:
Perhaps adding dependentFileTimes to BroadcastChange() in `InitialItemDataDoesNotUpdateLastItemsChangedAtUtc`

Reply:
I submitted another change similar to `InitialItemDataDoesNotUpdateLastItemsChangedAtUtc `